controllers/volumereplicationgroup_controller.go

```diff
@@ -538,65 +538,23 @@ const (
 	MModesLabel = "ramendr.openshift.io/maintenancemodes"
 )
 
-func (v *VRGInstance) processVRG() (ctrl.Result, error) {
+func (v *VRGInstance) processVRG() ctrl.Result {
 	if err := v.validateVRGState(); err != nil {
-		// record the event
-		v.log.Error(err, "Failed to validate the spec state")
-		rmnutil.ReportIfNotPresent(v.reconciler.eventRecorder, v.instance, corev1.EventTypeWarning,
-			rmnutil.EventReasonValidationFailed, err.Error())
-
-		msg := "VolumeReplicationGroup state is invalid"
-		setVRGDataErrorCondition(&v.instance.Status.Conditions, v.instance.Generation, msg)
-
-		if err = v.updateVRGStatus(false); err != nil {
-			v.log.Error(err, "Status update failed")
-			// Since updating status failed, reconcile
-			return ctrl.Result{Requeue: true}, nil
```

Comment:
Previously we handle failure in updateVRGStatus() by logging and requeuing. I don't see any logs in this case now. If this log is not needed please remove it a separate commit.
Making such changes with the current state of this code and the poor testing we have is risky.